src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs

```diff
@@ -1601,40 +1601,46 @@ private static bool IsAzureDevOpsArtifactsUrl(string location)
                     Directory.CreateDirectory(directory);
                 }
 
-                if (authHeader != null)
-                {
-                    requestMessage = new HttpRequestMessage(HttpMethod.Get, sourceUri)
-                    {
-                        Headers =
-                        {
-                            { HttpRequestHeader.Authorization.ToString(), authHeader.ToString() }
-                        }
-                    };
-                }
-                else
-                {
-                    requestMessage = new HttpRequestMessage(HttpMethod.Get, sourceUri);
-                }
-
                 // Ensure the parent target directory has been created.
                 using (FileStream outStream = new FileStream(temporaryFileName,
-                                                      _options.Overwrite ? FileMode.Create : FileMode.CreateNew,
-                                                      FileAccess.Write))
+                                                        _options.Overwrite ? FileMode.Create : FileMode.CreateNew,
+                                                        FileAccess.Write))
                 {
-                    using (var response = await client.SendAsync(requestMessage))
-                    {
-                        response.EnsureSuccessStatusCode();
-
-                        using (var inStream = await response.Content.ReadAsStreamAsync())
+                    await ExponentialRetry.Default.RetryAsync(
+                        async () =>
                         {
-                            downloadOutput.AppendLine($"    {sourceUri} =>");
-                            foreach (string targetFile in targetFiles)
+                            if (authHeader != null)
                             {
-                                downloadOutput.AppendLine($"      {targetFile}");
+                                requestMessage = new HttpRequestMessage(HttpMethod.Get, sourceUri)
+                                {
+                                    Headers =
```

Comment:
It feels like the iff could just be to add the header, there doesn't really seem to be a need to put the construction into different blocks (then you wouldn't need an "else" either, which would help with nesting a lot)

Reply:
@ChadNedzlek I believe the Headers property is get; only, so you can't add it later (hence the if/else block)

Reply:
This one? https://docs.microsoft.com/en-us/dotnet/api/system.net.http.headers.httprequestheaders.authorization?view=net-5.0#System_Net_Http_Headers_HttpRequestHeaders_Authorization

You wouldn't set the headers collection directly, you only set the properties you want.

Reply:
Yeah, this also is just moved/copied from above.

Reply:
Headers.Authorization has a setter. HttpRequestMessage.Headers does not. It only has a getter.

Reply:
Alright, I was able to do some shenanigans, and I think I've got this bit in a good place, giving us both the dispose, and allowing us to set Headers.Authorization as suggested. Let me know what you think